SDL: Fixed segfault for maps with empty tiles.

ablondin opened the pull request:
Hi!

Whenever I tried the included SDL example with a map having empty tiles, my application crashed. By looking more closely at the code, I noticed that the example does not take into account maps having empty tiles.

Since it's often hard to detect the origin of segfaults, I thought it would be a good idea to share the correction.

Any comments welcome!

baylej commented:
Hi,

I don't get why you have empty tiles, could you please post an example TMX map that reproduces the bug (you could use [gist](http://gist.github.com)), so then I can figure out if it's a bug in the example or in the library.

Many thanks for reporting!

ablondin commented:
I've added an example: https://gist.github.com/ablondin/19cc8760eb0e9cad5767.

It's an isometric map with 4 levels. It won't display correctly the tiles (since it's isometric and not rectangular, I'll have to tweak it), but the idea applies to rectangular maps if you have one layer that is not full. When I added the `if` checking for `NULL` tiles, everything worked fine, but it took me half an hour to figure out where the segfault was coming from.

The idea is that you might have some layers that are not fully covered by tiles (in my case, the layers correspond to different heights in the isometric tiling). On a rectangular map, I think your example would crash (though I did not test it) if there was any nonfull layer. Of course, I might be wrong :)

baylej commented:
This is indeed a bug in the examples (all of them) as the GID 0 means no tile.

ablondin commented:
I didn't take a look at the other examples, though, since I was only interested in your SDL example.

By the way, thanks for your library. My students will be using it in their practical homework (http://thales.math.uqam.ca/~blondin/fr/inf3135-tp2, it's in French, but I think you're from France)... If we find another bug, I'll let you know :)
